dotnet/src/Connectors/Connectors.Memory.PgVector/PostgresServiceCollectionExtensions.cs

```diff
@@ -1,163 +1,295 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
+using System;
 using Microsoft.Extensions.AI;
-using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.VectorData;
+using Microsoft.SemanticKernel;
 using Microsoft.SemanticKernel.Connectors.PgVector;
 using Npgsql;
 
-namespace Microsoft.SemanticKernel;
+namespace Microsoft.Extensions.DependencyInjection;
 
 /// <summary>
 /// Extension methods to register Postgres <see cref="VectorStore"/> instances on an <see cref="IServiceCollection"/>.
 /// </summary>
 public static class PostgresServiceCollectionExtensions
 {
     /// <summary>
-    /// Register a Postgres <see cref="VectorStore"/> with the specified service ID and where the NpgsqlDataSource is retrieved from the dependency injection container.
+    /// Register a <see cref="PostgresVectorStore"/> as <see cref="VectorStore"/>, where the <see cref="NpgsqlDataSource"/> is retrieved from the dependency injection container.
     /// </summary>
     /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="VectorStore"/> on.</param>
     /// <param name="options">Optional options to further configure the <see cref="VectorStore"/>.</param>
-    /// <param name="serviceId">An optional service id to use as the service key.</param>
+    /// <param name="lifetime">The service lifetime for the store. It needs to match <see cref="NpgsqlDataSource"/> lifetime. Defaults to <see cref="ServiceLifetime.Singleton"/>.</param>
     /// <returns>The service collection.</returns>
-    public static IServiceCollection AddPostgresVectorStore(this IServiceCollection services, PostgresVectorStoreOptions? options = default, string? serviceId = default)
+    public static IServiceCollection AddPostgresVectorStore(
```

Comment:
No Keyed variant of this (which doesn't accept a connection string)? I don't necessarily mind, just try to make sure we have a clear logic is all.

Reply:
That is correct, the methods that do resolve `NpgsqlDataSource` from DI don't have keyed overloads.

I just don't want to add too many overloads without having a clear need from the customers.

Reply:
Any reason to have them for SQLite but not here thouggh?